include/babeltrace/assert-internal.h

```diff
@@ -39,7 +39,7 @@
  */
 # define BT_ASSERT_FUNC
 #else
-# define BT_ASSERT(_cond)
+# define BT_ASSERT(_cond)	((void) sizeof((void) (_cond), 0))
```

Comment:
Briefly comment what this does above this line and put the link to <https://stackoverflow.com/questions/37411809/how-to-elegantly-fix-this-unused-variable-warning/37412551#37412551>.